packages/data-point/lib/entity-types/base-entity/factory.js

```diff
@@ -40,9 +87,67 @@ function create (Factory, spec, id) {
     entity.outputType = createReducer(outputType)
   }
 
-  entity.params = deepFreeze(defaultTo(spec.params, {}))
+  entity.params = defaultTo(spec.params, {})
+
+  return createEntityInstance(entity)
+}
+
+/**
+ * @param {Object} entity
+ * @returns {Object} entity named instance
+ */
+function createEntityInstance (entity) {
+  function EntityFactory () {}
+  Object.defineProperty(EntityFactory, 'name', {
+    value: entity.id
+  })
+
+  return deepFreeze(Object.assign(new EntityFactory(), entity))
+}
+
+/**
+ * @param {String} type Entity's type
+ * @param {Function} factory factory function to create the entity
+ * @param {Function} resolve function to resolve the entity instance
+ */
+function create (type, factory, resolve) {
+  if (arguments.length !== 3) {
+    throw new Error(
+      'Received wrong number of arguments. Try passing create(type:String, factory:Function, resolve:Function)'
+    )
+  }
+  if (typeof type !== 'string') {
+    throw new Error(
+      '"type" argument must be a string create(type:String, factory:Function, resolve:Function)'
+    )
+  }
+
+  validateFactory(factory)
+  validateResolve(resolve)
 
-  return entity
+  /**
+   * @param {String} name - Entity's name
+   * @param {Object} spec - spec for the Entity
+   * @returns {Object} Entity instance
+   */
+  return function createEntity (name, spec) {
```

Comment:
what was the reason for putting name before spec? curious, since putting "name" last would facilitate using a default param `(spec, name = 'generic')`,

Reply:
mainly readability when I did the tests and migrated code to use entity instances when the name was passed at the end, it was lost, and the name would usually be the same as the name of the variable, so it would be too far from each other. 

When I moved it as the first argument it was easier to read and to refactor, specially refactor